File: backend/src/main/java/com/sarahjewellers/model/EmailDraft.java
package com.sarahjewellers.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "email_drafts")
@Data
public class EmailDraft {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;

    private String subject;

    @Column(columnDefinition = "TEXT")
    private String body;

    private String occasion; // BIRTHDAY, ANNIVERSARY, FESTIVAL

    private String status; // PENDING, SENT

    @Column(name = "image_path")
    private String imagePath;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "sent_at")
    private LocalDateTime sentAt;
}

File: backend-node/models/EmailDraft.js
const mongoose = require('mongoose');

const emailDraftSchema = new mongoose.Schema({
  customer:  { type: mongoose.Schema.Types.ObjectId, ref: 'Customer', required: true },
  occasion:  { type: String, required: true },
  subject:   { type: String, required: true },
  body:      { type: String, required: true },
  status:    { type: String, enum: ['PENDING', 'SENT'], default: 'PENDING' },
  type:      { type: String, enum: ['DRAFT', 'BROADCAST'], default: 'DRAFT' },
  sentAt:    { type: Date },
  imagePath: { type: String }
}, { timestamps: true });

module.exports = mongoose.model('EmailDraft', emailDraftSchema);
